lib/src/main/java/com/shopify/checkoutkit/events/PixelEvent.kt

```diff
@@ -0,0 +1,1477 @@
+/*
+ * MIT License
+ *
+ * Copyright 2023-present, Shopify Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.shopify.checkoutkit.events
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.JsonObject
+
+@Serializable
+internal class PixelEventWrapper(
+    internal val name: String,
+    internal val event: JsonObject,
+)
+
+@Serializable
+public enum class EventType(public val typeName: String) {
+    @SerialName("dom") DOM("dom"),
+    @SerialName("standard") STANDARD("standard"),
+    @SerialName("custom") CUSTOM("custom");
+
+    public companion object {
+        public fun fromTypeName(typeName: String?): EventType? {
+            if (typeName == null) {
+                return null
+            }
+            return EventType.values().firstOrNull {
+                it.typeName == typeName
+            }
+        }
+    }
+}
+
+public interface PixelEvent {
```

Comment:
nit: if there is no scenario in which the SDK consumers will have to _implement_ this interface, we can make this a `sealed interface`. You keep the interface benefits while also preventing consumers from implementing this interface by mistake